apps/email/js/cards/lst/edit_controller.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+define(function(require, exports) {
+  var cards = require('cards'),
+      ConfirmDialog = require('confirm_dialog'),
+      deleteConfirmMsgNode = require('tmpl!../msg/delete_confirm.html'),
+      mozL10n = require('l10n!'),
+      toaster = require('toaster');
+
+  return {
+    createdCallback: function() {
+      this.editMode = false;
+      this.selectedMessages = null;
+      this.editModeEnabled = false;
+    },
+
+    setEditMode: function(editMode) {
+      // Do not bother if edit mode is not enabled yet.
+      if (!this.editModeEnabled) {
+        return;
+      }
+
+      this._setEditMode(editMode);
+    },
+
+    // This function is called from setEditMode() after ensuring that
+    // the backend is in a state where we can safely use edit mode.
+    _setEditMode: function(editMode) {
+      var i;
+
+      this.editMode = editMode;
+
+      // XXX the manual DOM play here is now a bit overkill; we should very
+      // probably switch top having the CSS do this for us or at least invest
+      // some time in cleanup.
+      if (editMode) {
+        this.classList.add('show-edit');
+
+        this.selectedMessages = [];
```

Comment:
Er, and note that despite my function name nitpicking in this area, I don't think we actually want to deal with the https://bugzilla.mozilla.org/show_bug.cgi?id=824001 issue yet since I think we do want the MailAPI to host a selection management abstraction that can avoid us having to manually deal with id's, and that can maybe help us with the statistics of how many messages in the set are starred/unread/etc. that we use to decide on UI button states.

(In fact, I'm tempted to implement it as basically a ViewSlice/ListView you can cram messages/conversations into and it keeps the messages in there updated.  This could be used in extensions for maintaining temporary stacks of messages to deal with for to-do lists/etc.)